src/controller/CHIPDeviceController.cpp

```diff
@@ -1475,20 +1474,25 @@ void BasicFailure(void * context, uint8_t status)
 {
     ChipLogProgress(Controller, "Received failure response %d\n", (int) status);
     DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);
-    commissioner->OnSessionEstablishmentError(static_cast<CHIP_ERROR>(status));
+    commissioner->CommissioningStageComplete(static_cast<CHIP_ERROR>(status));
 }
 
-void DeviceCommissioner::CommissioningStageComplete(CHIP_ERROR err)
+void DeviceCommissioner::CommissioningStageComplete(CHIP_ERROR err, CommissioningDelegate::CommissioningReport report)
 {
     if (mCommissioningDelegate == nullptr)
     {
         return;
     }
-    CommissioningDelegate::CommissioningReport report(mCommissioningStage);
-    CHIP_ERROR status = mCommissioningDelegate->CommissioningStepFinished(err, report);
+    report.stageCompleted = mCommissioningStage;
+    CHIP_ERROR status     = mCommissioningDelegate->CommissioningStepFinished(err, report);
     if (status != CHIP_NO_ERROR)
     {
-        OnSessionEstablishmentError(status);
+        // Commissioning delegate will only return error if it failed to perform the appropriate commissioning step.
+        // In this case, we should call back the commissioning complete and call session error
+        if (mPairingDelegate != nullptr)
+        {
+            mPairingDelegate->OnCommissioningComplete(mDeviceBeingCommissioned->GetDeviceId(), status);
```

Comment:
OnCommissioningComplete drops the NodeID to the floor. It seems that we should provide a full operational ID on OnCommissioningComplete, not just NodeID, if we pass anything. Right now, all in-SDK users disregard the NodeID, adn a NodeID without Root public key and fabric ID is ambiguous, so we should probably avoid passing it.

Reply:
Fair enough - can address in a follow up. When this was written, node id was king. https://github.com/project-chip/connectedhomeip/issues/14357